### PR TITLE
Changed block size calculation in Amgcl_solver

### PR DESCRIPTION
**Pull request by roigcarlo**
The missing changes we spoke in the meeting.

### CODE REVIEW

File: kratos/linear_solvers/amgcl_solver.h [hunk 1]
```diff
@@ -543,39 +543,70 @@ class AMGCLSolver : public LinearSolver< TSparseSpaceType,
...
-                    if(old_ndof == -1) old_ndof = ndof;
-                    else if(old_ndof != ndof) { //if it is different than the block size is 1
-                        old_ndof = -1;
-                        break;
-                    }
 
-                    ndof=1;
-                } else {
-                    ndof++;
+        if(rModelPart.GetCommunicator().GetDataCommunicator().IsDistributed() == false)
```
Comment (philbucher) [inline kratos/linear_solvers/amgcl_solver.h]:
```suggestion
        if (!rModelPart.IsDistributed())
```

Comment (roigcarlo) [reply kratos/linear_solvers/amgcl_solver.h]: Oo Can this be called directly without the communicator?

Comment (philbucher) [reply kratos/linear_solvers/amgcl_solver.h]: [Yep](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/includes/model_part.h#L1707-L1710)

**Comment by roigcarlo** [comment]
Ups, I though I uploaded the sign fix. Thx @ddiezrod !!

**Comment by ddiezrod** [comment]
You're wlcome ;) can you take a look @RiccardoRossi ?

**Review by RiccardoRossi — APPROVED** [review]

**Pull request merged**